### PR TITLE
Upgrade to create-react-app v5 and react v17

### PR DESCRIPTION
**Pull request by alanwguo**
Signed-off-by: Alan Guo <aguo@anyscale.com>

## Why are these changes needed?
Upgrade to create-react-app v5 and react v17
Update typescript to version 4.8
Update to use latest recommended lint rules

Follow-ups:
1. Upgrade to react 18
2. Upgrade to mui v5

## Related issue number
fixes #29537

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
What's the implication of the update? Is there anything like unsupported platform or something like that?

**Comment by rkooo567** [comment]
Screen Shot 2022-10-25 at 5 27 24 AM
this may be related?

**Comment by alanwguo** [comment]
> What's the implication of the update? Is there anything like unsupported platform or something like that?

Forces people to have a newer version of node and npm installed. The create-react-app package is primarily a build tool so shouldn't impact end user.

React 17 is known to not deprecate any previous browser versions

**Comment by rkooo567** [comment]
sgtm. Lmk when to merge this! 

> Forces people to have a newer version of node and npm installed. The create-react-app package is primarily a build tool so shouldn't impact end user.

Is this not the case for users, but only for building wheels?

**Comment by alanwguo** [comment]
@rkooo567 , this looks good to merge!

**Comment by rkooo567** [comment]
We need a doc update. cc @ericl @richardliaw for the approval. Also @alanwguo do you think we need review from other people from your team?

File: dashboard/client/package.json [hunk 1]
```diff
@@ -62,18 +66,32 @@
...
-      "prefer-arrow"
+      "prefer-arrow",
+      "@typescript-eslint"
     ],
     "rules": {
       "@typescript-eslint/consistent-type-definitions": [
         "error",
         "type"
       ],
+      "@typescript-eslint/no-explicit-any": 0,
```
Comment (shawnpanda) [inline dashboard/client/package.json:84]: sad sad. Are there a lot of `any` in the codebase?

**Review by shawnpanda — APPROVED** [review]

File: dashboard/client/package.json (hunk 1 above)
Comment (alanwguo) [reply dashboard/client/package.json:84]: yes there are a few. I think this is fine. Sometimes, having the "any" escape hatch will be necessary when dealing with third-party libraries.

At least implicit anys are still banned, so if someone is using any, they have to very explicitly set it in the code.

**Comment by alanwguo** [comment]
@shawnpanda bump

**Comment by alanwguo** [comment]
@ericl , can i get a docs review on this?

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**